src/util/bip32.rs

```diff
@@ -261,19 +268,36 @@ impl FromStr for DerivationPath {
 }
 
 impl DerivationPath {
-    /// Create a new DerivationPath that is a child of this one.
+    /// Create a new [DerivationPath] that is a child of this one.
     pub fn child(&self, cn: ChildNumber) -> DerivationPath {
         let mut path = self.0.clone();
         path.push(cn);
         DerivationPath(path)
     }
 
-    /// Convert into a DerivationPath that is a child of this one.
+    /// Convert into a [DerivationPath] that is a child of this one.
     pub fn into_child(self, cn: ChildNumber) -> DerivationPath {
         let mut path = self.0;
         path.push(cn);
         DerivationPath(path)
     }
+
+    /// Returns the [DerivationPath] that is a single increment from this one.
+    pub fn increment(&self) -> Result<DerivationPath, Error> {
```

Comment:
Is the increment of a path well-defined? I couldn't find it in the standard and just incrementing the last part seems a bit arbitrary (although I see the use case). Maybe a child iterator that starts at a certain child number and yields all following children would solve the problem too with the added benefit of access to the iterator API? Then a wallet could do the following to calculate its balance:

```rust
let first_normal = ChildNumber::from_normal_idx(0).unwrap();
let path: DerivationPath = "m/0/0'".parse().unwrap();

let balance = path.children_since(first_normal)
    .flat_map(get_utxos_for_path)
    .map(utxo_amount)
    .sum();
```

Reply:
+1 to having an iterator instead

Reply:
Yeah that makes sense. Take a look at the new push.